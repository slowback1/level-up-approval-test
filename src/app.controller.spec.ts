import { Test, TestingModule } from '@nestjs/testing';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { configure } from "approvals";
import { JestReporter } from "approvals/lib/Providers/Jest/JestReporter";
import {verify, verifyAsJson} from "approvals/lib/Providers/Jest/JestApprovals";

describe('AppController', () => {
  let appController: AppController;
  beforeAll(() => {
    configure({
      reporters: [new JestReporter()],
    });
  });

  beforeEach(async () => {
    const app: TestingModule = await Test.createTestingModule({
      controllers: [AppController],
      providers: [AppService],
    }).compile();

    appController = app.get<AppController>(AppController);
  });

  describe('root', () => {
    it('should return "Hello World!"', () => {
      verify(appController.helloWorld());
    });
    it("has an approved output for the get pokemon endpoint", () => {
      verifyAsJson(appController.getPokemon());
    })
  });
});
