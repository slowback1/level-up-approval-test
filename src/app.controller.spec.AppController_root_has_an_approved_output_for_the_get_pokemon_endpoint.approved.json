[
  {
    "height": 7,
    "id": 1,
    "name": "bulbasaur",
    "base_experience": 64,
    "types": [
      "grass",
      "poison"
    ]
  },
  {
    "height": 10,
    "id": 2,
    "name": "ivysaur",
    "base_experience": 142,
    "types": [
      "grass",
      "poison"
    ]
  },
  {
    "height": 20,
    "id": 3,
    "name": "venusaur",
    "base_experience": 263,
    "types": [
      "grass",
      "poison"
    ]
  },
  {
    "height": 6,
    "id": 4,
    "name": "charmander",
    "base_experience": 62,
    "types": [
      "fire"
    ]
  },
  {
    "height": 11,
    "id": 5,
    "name": "charmeleon",
    "base_experience": 142,
    "types": [
      "fire"
    ]
  },
  {
    "height": 17,
    "id": 6,
    "name": "charizard",
    "base_experience": 267,
    "types": [
      "fire",
      "flying"
    ]
  },
  {
    "height": 5,
    "id": 7,
    "name": "squirtle",
    "base_experience": 63,
    "types": [
      "water"
    ]
  },
  {
    "height": 10,
    "id": 8,
    "name": "wartortle",
    "base_experience": 142,
    "types": [
      "water"
    ]
  },
  {
    "height": 16,
    "id": 9,
    "name": "blastoise",
    "base_experience": 265,
    "types": [
      "water"
    ]
  },
  {
    "height": 3,
    "id": 10,
    "name": "caterpie",
    "base_experience": 39,
    "types": [
      "bug"
    ]
  },
  {
    "height": 7,
    "id": 11,
    "name": "metapod",
    "base_experience": 72,
    "types": [
      "bug"
    ]
  },
  {
    "height": 11,
    "id": 12,
    "name": "butterfree",
    "base_experience": 198,
    "types": [
      "bug",
      "flying"
    ]
  },
  {
    "height": 3,
    "id": 13,
    "name": "weedle",
    "base_experience": 39,
    "types": [
      "bug",
      "poison"
    ]
  },
  {
    "height": 6,
    "id": 14,
    "name": "kakuna",
    "base_experience": 72,
    "types": [
      "bug",
      "poison"
    ]
  },
  {
    "height": 10,
    "id": 15,
    "name": "beedrill",
    "base_experience": 178,
    "types": [
      "bug",
      "poison"
    ]
  },
  {
    "height": 3,
    "id": 16,
    "name": "pidgey",
    "base_experience": 50,
    "types": [
      "normal",
      "flying"
    ]
  },
  {
    "height": 11,
    "id": 17,
    "name": "pidgeotto",
    "base_experience": 122,
    "types": [
      "normal",
      "flying"
    ]
  },
  {
    "height": 15,
    "id": 18,
    "name": "pidgeot",
    "base_experience": 216,
    "types": [
      "normal",
      "flying"
    ]
  },
  {
    "height": 3,
    "id": 19,
    "name": "rattata",
    "base_experience": 51,
    "types": [
      "normal"
    ]
  },
  {
    "height": 7,
    "id": 20,
    "name": "raticate",
    "base_experience": 145,
    "types": [
      "normal"
    ]
  }
]
